### ISSUE TITLE
Feature request: A callback for when toolbar visibility has changed.

### ISSUE THREAD

**Issue by jwehrle** [memspace/zefyr#213]
I needed this to remove some padding/margins when the toolbar was being shown so I added the following to zefyr/scr/widgets/scaffold.

    typedef OnZefyrVisibilityChanged = void Function(bool isVisible);
    class ZefyrScaffold extends StatefulWidget {
      final Widget child;
      final OnZefyrVisibilityChanged onZefyrVisibilityChanged;

      const ZefyrScaffold({Key key, this.child, this.onZefyrVisibilityChanged})
          : super(key: key);

      static ZefyrScaffoldState of(BuildContext context) {
        final _ZefyrScaffoldAccess widget =
            context.inheritFromWidgetOfExactType(_ZefyrScaffoldAccess);
        return widget.scaffold;
      }

     @override
      ZefyrScaffoldState createState() => ZefyrScaffoldState();
    }

And then in the state...

    void showToolbar(WidgetBuilder builder) {
    setState(() {
      _toolbarBuilder = builder;
      if (widget.onZefyrVisibilityChanged != null) {
        widget.onZefyrVisibilityChanged(true);
      }
    });
      }

    void hideToolbar() {
    if (_toolbarBuilder != null) {
      setState(() {
        _toolbarBuilder = null;
        if (widget.onZefyrVisibilityChanged != null) {
          widget.onZefyrVisibilityChanged(false);
        }
      });
    }
      }

**Comment by pulyaevskiy** [comment]
You probably could simply listen for updates on the FocusNode you provide for editor. Toolbar is only visible when the editor has focus.

Though toolbar itself gains focus sometimes (for links) so it’d not work in this case...

Just thinking out loud...

**Comment by jwehrle** [comment]
Yes, the link issue is the one that was causing a problem. As a work around I had another textfield with its own focusnode and so listening to both I was able to work out when the ZefyEditor had lost focus because the toolbar had it.
But, all things considered, it would have been easier and more reliable if there were simply a callback for this.
